src/treemap/split.js

```diff
@@ -0,0 +1,105 @@
+export default function(parent, x0, y0, x1, y1) {
+
+  parent.x0 = x0;
+  parent.x1 = x1;
+  parent.y0 = y0;
+  parent.y1 = y1;
+
+  // exit if no children
+  if (!(parent.children) || parent.children.length === 0) {
+    return;
+  }
+
+  splitTree(parent.children, parent.value, x0, y0, x1, y1);
```

Comment:
This looks like it assumes that parent.value is equal to the sum of the children’s values, which won’t be the case if the parent has a self-value. You’ll need to sum parent.children[i].value before calling splitTree.